orchagent/portsorch.cpp

```diff
@@ -1634,7 +1634,10 @@ void PortsOrch::doPortTask(Consumer &consumer)
             {
                 for (auto it = m_portListLaneMap.begin(); it != m_portListLaneMap.end();)
                 {
-                    if (m_lanesAliasSpeedMap.find(it->first) == m_lanesAliasSpeedMap.end())
+                    sai_uint32_t port_speed;
+                    auto is = m_lanesAliasSpeedMap.find(it->first);
+                    getPortSpeed(it->second, port_speed);
+                    if ((m_lanesAliasSpeedMap.find(it->first) == m_lanesAliasSpeedMap.end()) || (get<1>(is->second) != port_speed))
```

Comment:
why do we need to compare the port speed here?

Reply:
When number of lanes are same but speed is different in port_config.ini(and config_db.json), we still need to remove port and add port. Without this change the port will not get recreated for new speed.

Reply:
if lanes are the same set, then we just need to set speed. why do we need to remove and add port?

Reply:
When we breakout a port from one 100G (2lane * 50G PAM4) to two 50G (2lane * 25G NRZ) then we will have to recreate the port as the FEC mode could change. Same is true when we breakout a port from one 200G (4lane * 50G PAM4) to two 100G (4lane * 25G NRZ).

Reply:
sai should handle the fec change accordingly, or let application handle the fec settings. such recreating port should not be expose to application.

Reply:
I removed this change